lib/app/ctx.js

```diff
@@ -224,6 +224,10 @@ Box.prototype.unwrap = function (data, content) {
             if (msg.type.substr(0, 3) === 'QLB' && tr.docs.length > 0) {
                 content = tr.docs.shift().contents;
             }
+            if (msg.type === 'DOCUMENT') {
+                data = msg.contents;
```

Comment:
Inside "data" we have a signed data (USC_SIGN QLB_SIGN), and that signature will be verified at the next loop.
All other items from XML are saved in the headers, and they are not overwritten later. So we do not lose anything, as I see.
Am I wrong?